src/annotator/anchoring/test/range-test.js

```diff
@@ -0,0 +1,532 @@
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.deepEqual(normalizedRange, createNormalizedRange());
+        });
+
+        it('handles ELEMENT_NODE start node when `startOffset` > than number of child nodes', () => {
+          const browserRange = createBrowserRange({
+            startContainer: container.querySelector('#p-1'),
+            startOffset: 1,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.deepEqual(normalizedRange, createNormalizedRange());
+        });
+      });
+
+      context('endContainer is ELEMENT_NODE', () => {
+        it('handles ELEMENT_NODE end node', () => {
+          const browserRange = createBrowserRange({
+            endContainer: container.querySelector('#p-3'),
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.deepEqual(normalizedRange, createNormalizedRange());
+        });
+
+        it('handles ELEMENT_NODE end node with no `endOffset` and no children', () => {
+          const browserRange = createBrowserRange({
+            endContainer: container.querySelector('#p-4'),
+            endOffset: 0,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.deepEqual(normalizedRange, createNormalizedRange());
+        });
+
+        it('handles ELEMENT_NODE end nodes with no `endOffset` and children', () => {
+          const browserRange = createBrowserRange({
+            endContainer: container.querySelector('#p-3'),
+            endOffset: 0,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.deepEqual(normalizedRange, createNormalizedRange());
+        });
+
+        it('handles ELEMENT_NODE end nodes with no `endOffset` and non-TEXT_NODE children', () => {
+          const browserRange = createBrowserRange({
+            endContainer: container.querySelector('#p-5'),
+            endOffset: 0,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.deepEqual(normalizedRange, createNormalizedRange());
+        });
+      });
+
+      context('slices the text elements', () => {
+        it('cuts the text node if there is no sibling', () => {
+          const browserRange = createBrowserRange({
+            startOffset: 1,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.equal(normalizedRange.start.data, 'ext 1');
+        });
+
+        it('cuts the text node if there is a sibling but text node length > `startOffset`', () => {
+          const browserRange = createBrowserRange({
+            startContainer: container.querySelector('#p-2').firstChild,
+            startOffset: 1,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.equal(normalizedRange.start.data, 'ext 2a');
+        });
+
+        it('does not cut the text node', () => {
```

Comment:
```suggestion
        it('does not cut the text node', () => {
```

Shouldn't there be a "when <condition>" at the end of this test description?

Reply:
yes, I noticed a mistake in a previous test description too, fixed both.